x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlIndexTemplateRegistry.java

```diff
@@ -125,15 +124,14 @@ private static IndexTemplateConfig statsTemplate(boolean ilmEnabled) {
     public MlIndexTemplateRegistry(Settings nodeSettings, ClusterService clusterService, ThreadPool threadPool, Client client,
                                    NamedXContentRegistry xContentRegistry) {
         super(nodeSettings, clusterService, threadPool, client, xContentRegistry);
-        boolean ilmEnabled = XPackSettings.INDEX_LIFECYCLE_ENABLED.get(settings);
         templatesToUse = Arrays.asList(
             ANOMALY_DETECTION_RESULTS_TEMPLATE,
-            ilmEnabled ? ANOMALY_DETECTION_STATE_TEMPLATE : ANOMALY_DETECTION_STATE_TEMPLATE_NO_ILM,
+            ANOMALY_DETECTION_STATE_TEMPLATE,
```

Comment:
Can we remove `ANOMALY_DETECTION_STATE_TEMPLATE_NO_ILM` and `STATS_TEMPLATE_NO_ILM` now?

Reply:
I will reach out to someone from ML to see if this change is OK, but I would guess that we will be able to remove those templates.

- [ ] get feedback from someone on ML for removing `NO_ILM` templates

Reply:
@dimitris-athanasiou and @przemekwitek have confirmed that these templates can be removed.

Reply:
(Przemek from ML team): I think it is ok to remove the non-ILM versions.
Once only ILM versions exist, it would be good to inline the variables into the .json files but if you don't feel like doing it, I can do it as a followup PR.

Reply:
@przemekwitek Do you mean like this?

```
    private static IndexTemplateConfig stateTemplate() {
        Map<String, String> variables = new HashMap<>();
        variables.put(VERSION_ID_PATTERN, String.valueOf(Version.CURRENT.id));
        variables.put("xpack.ml.index.lifecycle.name", ML_SIZE_BASED_ILM_POLICY_NAME);
        variables.put("xpack.ml.index.rollover_alias.name", AnomalyDetectorsIndex.jobStateIndexWriteAlias());

        return new IndexTemplateConfig(AnomalyDetectorsIndexFields.STATE_INDEX_PREFIX,
            ANOMALY_DETECTION_PATH + "state_index_template.json",
            Version.CURRENT.id, VERSION_PATTERN,
            variables);
    }
```

…with this in the template json file?

```
  "settings" : {
    "index" : {
      "auto_expand_replicas" : "0-1",
      "hidden": true
    },
    "index.lifecycle.name": "${xpack.ml.index.lifecycle.name}",
    "index.lifecycle.rollover_alias": "${xpack.ml.index.rollover_alias.name}"
  },
```

Or should the actual values be inlined as well?

Reply:
Any of these 2 solutions work for me. 
It's most important for me to get rid of syntax tokens (like comma ",") which were just a hack to allow enabled/disabled duality.
The first one is more flexible so my preference goes there.

Reply:
I've pushed a commit with the first, more flexible strategy. Thanks for the feedback!

Reply:
Could you make the variable name the same for both templates?

Reply:
Good catch. This is done.

Reply:
Thank you!